src/library/coverartdelegate.cpp

```diff
@@ -108,17 +107,14 @@ void CoverArtDelegate::paintItem(QPainter *painter,
     info.hash = index.sibling(index.row(), m_iCoverHashColumn).data().toUInt();
     info.trackLocation = index.sibling(index.row(), m_iTrackLocationColumn).data().toString();
 
+    double scaleFactor = getDevicePixelRatioF(dynamic_cast<QWidget*>(parent()));
     // We listen for updates via slotCoverFound above and signal to
     // BaseSqlTableModel when a row's cover is ready.
-    QPixmap pixmap = pCache->requestCover(info, this, option.rect.width(),
+    QPixmap pixmap = pCache->requestCover(info, this, option.rect.width() * scaleFactor,
                                           m_bOnlyCachedCover, true);
     if (!pixmap.isNull()) {
-        int width = math_min(pixmap.width(), option.rect.width());
-        int height = math_min(pixmap.height(), option.rect.height());
-        QRect target(option.rect.x(), option.rect.y(),
-                     width, height);
-        QRect source(0, 0, target.width(), target.height());
-        painter->drawPixmap(target, pixmap, source);
+        pixmap.setDevicePixelRatio(scaleFactor);
```

Comment:
Original I think the code prevents us from 
scale up a small source cover for a better view. The issue is that scaling a cover by let's say 1.3 will look blurry. Original the unscaled cover was shown.
We can consider here rounding the scale factor to the next good looking.
What do you think.

Reply:
But that's the deal with the HiDPI mode. Otherwise bitmap graphics would appear too small compared to the remaining elements or widgets. Applying the scaling only to vectorized but not to bitmap graphics would cause inconsistent sizing.

Reply:
The issue here is slightly different, because the cover bitmaps are not made for our purpose.

For example if we have a 16x16 source bitmap and the column width is 19.on a 200 % HiDPi screen. 

This code will render the Cover with 38 px which is a scale factor 2.375 which looks kind of blurry.

The original code would take the integer 2 for 32 px + 3 pixel border.
This will look much better.

The blurryness solved by this PR if the source cover is already > 38. Here we can also improve the results with conditional borders, but IMHO the effect is less visible and can be ignored.

Reply:
> The blurryness solved by this PR if the source cover is already > 38.

I have never come across a cover image so small, so I think this discussion is largely hypothetical.

Reply:
I have picked the small size only as example. You have the same blurryness if you for example show a 1600 x 1920 picture on a 1080 x 1920 screen. 
It is the case if both ratios have no comon determinator. 

Here is an example of a 250 px cover scaled to 255.
A ratio of 1.02. The left one is the scaled one, the right one has a border.
Be sure to watch it at 100 % to see the difference.
![cover](https://user-images.githubusercontent.com/1777442/68289207-02c05c00-0086-11ea-9cc5-c7e26e37a4fe.png)

Original we have special cased for the right. Removing this is a regression, we need to decide if we can effort it.

Reply:
This issue does'nt matter much in the practice, so I think it is OK to ignore it.